page-tracking/src/page_info.rs

```diff
@@ -159,6 +159,24 @@ impl PageInfo {
 
     /// Pops the current owner if there is one, returning the page to the previous owner.
     pub fn release(&mut self) -> PageTrackingResult<()> {
```

Comment:
nit: i think it'd be slightly cleaner if you had a single release with a `was_invalid: bool` param that was used to determine the behavior in the `BlockedShared` state, rather than having `release()` fall through to `release_invalidated()`.